plugins/SegmentEditor/templates/_segmentSelector.twig

```diff
@@ -1,5 +1,5 @@
 <div class="SegmentEditor" style="display:none;">
-    <div class="segmentationContainer listHtml" title="{{ 'SegmentEditor_ChooseASegment'|translate|e('html_attr') }}. {{ 'SegmentEditor_CurrentlySelectedSegment'|translate(segmentDescription)|e('html_attr') }}">
```

Comment:
Any reason for removing that additional escaping?

Reply:
It was just a find/replace over multiple files. I noticed when looking at the html source of files that were doing `json_encode|e('html_attr')` that it actually encodes way more than it needs to. Seems like html_attr is used for unquoted html attrs, eg, `myattr={{ myvalue|e('html_attr') }}`. It just adds to the byte count unnecessarily, especially when used with json_encode. Since these are translations it probably has no effect here, so this could be reverted, but there's no real reason to keep it.

Reply:
We should keep the escaping where no json_encode is already done. The translation might actually contain a `"` in other languages, which would cause the HTML to become invalid.

Reply:
@sgiehl Something about this doesn't make sense, json_encode would result in `"` too, no? ie, `{"abc": "def"}`. I will look into this further later. I guess its a small issue overall if its safe to remove the filter, but I'd like to be clear in what's going on underneath here. At least for my own edification.

Reply:
@sgiehl did a test with this code:

```
$v = new Piwik\View('@CoreHome/testFile.twig');
$v->str = "a value with strange chars <&\"'{}>/\\";
$v->obj = [
    'key' => ['val1', 'val2', 'val3'],
    'key2' => 'val3"with"quot&e&s',
];
$res = $v->render();

print $res;
print "\n";
```

and this template:

```

with nothing:

STR: {{ str }}
OBJ: {{ obj|json_encode }}

with e html:

STR: {{ str|e('html') }}
OBJ: {{ obj|json_encode|e('html') }}

with e html_attr:

STR: {{ str|e('html_attr') }}
OBJ: {{ obj|json_encode|e('html_attr') }}

with \{\{-

STR: {{- str -}}
OBJ: {{- obj|json_encode -}}

with set:

{% set out %}
STR: {{ str }}
OBJ: {{ obj|json_encode }}
{% endset %}

{{ out }}
```

this is the output I got:

```

with nothing:

STR: a value with strange chars &lt;&amp;&quot;&#039;{}&gt;/\
OBJ: {&quot;key&quot;:[&quot;val1&quot;,&quot;val2&quot;,&quot;val3&quot;],&quot;key2&quot;:&quot;val3\&quot;with\&quot;quot&amp;e&amp;s&quot;}

with e html:

STR: a value with strange chars &lt;&amp;&quot;&#039;{}&gt;/\
OBJ: {&quot;key&quot;:[&quot;val1&quot;,&quot;val2&quot;,&quot;val3&quot;],&quot;key2&quot;:&quot;val3\&quot;with\&quot;quot&amp;e&amp;s&quot;}

with e html_attr:

STR: a&#x20;value&#x20;with&#x20;strange&#x20;chars&#x20;&lt;&amp;&quot;&#x27;&#x7B;&#x7D;&gt;&#x2F;&#x5C;
OBJ: &#x7B;&quot;key&quot;&#x3A;&#x5B;&quot;val1&quot;,&quot;val2&quot;,&quot;val3&quot;&#x5D;,&quot;key2&quot;&#x3A;&quot;val3&#x5C;&quot;with&#x5C;&quot;quot&amp;e&amp;s&quot;&#x7D;

with \{\{-

STR:a value with strange chars &lt;&amp;&quot;&#039;{}&gt;/\OBJ:{&quot;key&quot;:[&quot;val1&quot;,&quot;val2&quot;,&quot;val3&quot;],&quot;key2&quot;:&quot;val3\&quot;with\&quot;quot&amp;e&amp;s&quot;}with set:


STR: a value with strange chars &lt;&amp;&quot;&#039;{}&gt;/\
OBJ: {&quot;key&quot;:[&quot;val1&quot;,&quot;val2&quot;,&quot;val3&quot;],&quot;key2&quot;:&quot;val3\&quot;with\&quot;quot&amp;e&amp;s&quot;}

```

As you can see, e|html_attr seems to behave exactly the same as e|html except it also encodes far more characters, including spaces. I believe its to allow html attributes to be output without quotes (though why anyone would WANT to do this is beyond me). Does this make sense to you? Am I missing something?

Reply:
Note: in this test the autoescape is working, so I'm not sure why it isn't in the other code...

Reply:
I had actually never spent too many thoughts on the twig escaping stuff and thought it would make sense to always add `|e('html_attr')` for variables in html attributes. But seem that actually really doesn't seem to make that much sense 🤷 
So guess it should be fine to remove it, if we can rely on the autoescaping...

Reply:
I did too, I just accidentally noticed `|e('html_attr')` was adding a lot of extra bytes by looking at a page w/ view-source: and only because I was adding so many `|json_encode|e('html_attr')`s. Wouldn't have noticed otherwise.